ocaml/src/osd_access.ml

```diff
@@ -357,11 +357,58 @@ class osd_access
       make_alba_osd_client
   in
 
+  let categorise_exception osd_info =
+    let open Asd_protocol.Protocol.Error in
+    function
+    (* -> should_invalidate_pool, should_retry, add_to_errors aka disqualify  *)
+    | Exn Assert_failed _   -> false, false, false
+    | Exn Unknown_operation -> false, false, true
+    | Exn Full              -> true,  false, true
+       (* when it's full, we need to disqualify this OSD,
+          and the 'set' in the requalification loop will eventually requalify it,
+          fe, when rebalancing creates space on that OSD again
+        *)
+    | End_of_file                            -> true,  true, false
+    | Unix.Unix_error(Unix.ECONNRESET,_,_)   -> true, false, true
+    | Unix.Unix_error(Unix.EPIPE,_,_)        -> true, false, true
+
+    | Unix.Unix_error(Unix.ECONNREFUSED,_,_) -> false, false, true
+    | (Client_helper.MasterLookupResult.Error t) as exn ->
+       let long_id =
+         let open OsdInfo in
+         get_long_id osd_info.kind in
+       let () =
+         Lwt_log.ign_debug_f
+           "%s %S : %s"
+           long_id
+           (Client_helper.MasterLookupResult.to_string t)
+           (Printexc.to_string exn)
+       in
+       false, true, false
+    | exn -> (* by experience: don't be so optimistic *)
+       begin
+         let open OsdInfo in
+         let long_id = get_long_id osd_info.kind in
+         let () =
+           Lwt_log.ign_info_f
+             "%s %S was unforeseen"
+             long_id
+             (Printexc.to_string exn)
+         in
+         match osd_info.kind with
+           | Asd     _   -> true, false, true
+           | Kinetic _   -> true, false, true
+           | Alba    _   -> true, false, false
+           | Alba2   _   -> true, false, false
```

Comment:
this should be `true, false, true` for all kinds of osds?

Reply:
I think you want don't want to be that pessimistic for Alba & Alba2 osds

Reply:
ok.